File: 002/002.hs
fib 0 = 0
fib 1 = 1
fib x = let f = \a b c -> if (c - 1) == 1
                          then a + b
                          else f b (a + b) (c - 1)
        in f 0 1 x

fibSum = sum $ takeWhile (< 4000000) $ map fib [3,6..]

main = putStrLn $ show fibSum

File: 002/002.clj
(defn fib [n]
  (cond
    (= n 0) 0
    (= n 1) 1
    :else (loop [x 0 y 1 z n]
            (if (= (dec z) 1)
              (+ x y)
              (recur y (+ x y) (dec z))))))

(defn fib-sum []
  (apply + 
    (filter even? 
      (take-while (partial > 4000000)
        (map fib (range))))))

(println (fib-sum))
